docs/vgclog.md

```diff
@@ -0,0 +1,423 @@
+
+- **`timestamp`**, **`id`**, and **`context-id`** Each event that is logged in the GC cycle stanza is labeled with a `timestamp`,`id`, and `context-id` XML attribute. The `id` attribute increases incrementally with each event. All gc increments, operations, and concurrent tags associate with a particular cycle have a `contxtid` value that matches the `id` value of the cycle. 
+
+- **`<gc-start>`** The `<gc-start>` tag logs the beginning of a GC increment, in this case of type `scavenge`, nested within the GC cycle. 
+
+- **`<gc-op>`** Nested within the GC increment is the `<gc-op>` tag, which logs information about the individual GC ops that make up the GC increment. In this case, only one operation, the scavenge operation, makes up the GC increment, and only one GC increment makes up this *stop-the-world* cycle. In particular, a scavenge operation evacuates the local area of the heap in a single operation and so a single operation makes up the whole GC cycle. 
+ A GC algorithm can also involve multiple GC operations, such as a combination of mark, sweep, compact, or copy. For a common GC of mark-sweep, the GC increment consists of two operations, mark and sweep. This increment might be accompanied by other GC increments to make up a GC cycle that reclaims memory.
+
+- **`'type'`** *Stop-the-world* and concurrent cycles can be solely local or global events, or can consist of a mixture of local and global operations. You can determine whether the cycles, increments, or operations are local or global from the XML attribute `“type”`. The gencon policy has two types of GC collection; global concurrent, and scavenge, which is a local operation.
+
+
+<!-- Add link to topics for analysing logs and tag definitions once they have been created -->
+
+## Using the verbose GC log to troubleshoot
+
+### *Stop-the-world* cycles and concurrent cycles 
+
+Garbage collectors run collections as either a *stop-the-world* cycle, or as a concurrent cycle. For example, the gencon policy by default consists of the following collectors:
+
+- a [scavenger collector](gc.md#garbage-collection-policies), which uses a *stop-the-world* cycle to collect.
+
+- a [concurrent global mark and sweep collector](gc.md#garbage-collection-policies), which uses a concurrent cycle to collect.  
+
+The first step in reading the verbose GC log is to understand how these different types of cycle are recorded. 
+
+Verbose GC logs only record *stop-the-world* events, which are events that involve pauses of the application threads of the VM. *Stop-the-world* cycles consist solely of *stop-the-world* events so all events of a *stop-the-world* cycle are recorded in the verbose GC logs. For concurrent cycles, while some operations of concurrent cycles run at the same time as application threads, others run as *stop-the-world* events. In particular, concurrent cycles consist of: 
```

Comment:
Still some confusing language around what is and what is not a STW event. If I understand correctly, everything recorded in the log is an `<exclusive-start>` & therefore a STW event, it's just that some are part of a global collection and some are part of a concurrent collection (marked by `<concurrent-kickoff>`)? Are we maybe using the term *stop-the-world* cycle when we mean Global GC cycle?

Reply:
There are 2 things to know:

- Global vs. local is independent of whether the GC cycle is a stop-the world cycle or concurrent cycle. 

- There are 2 types of cycle to think about:
1. A stop-the-world cycle, which consists of only stop-the-world events.  
2. A concurrent cycle, which is made up of both stop-the-world events and concurrent events. 

I'll take a look at the topic to improve how this is explained.